Git Repository fails

### Environment
* Python version:  <!-- Example: 3.7.7 -->
* Nautobot version:  <!-- Example: 1.2.0 -->
* design-builder version:  <!-- Example: 0.1.0 -->

<!-- What did you expect to happen? -->
### Expected Behavior

Creation of Git Repository fails

<!-- What happened instead? -->
### Observed Behavior

Creation of Git Repository is possible

<!--
    Describe in detail the exact steps that someone else can take to reproduce
    this bug using the current release.
-->
### Steps to Reproduce
1. Create Git Repository2. 
```
git_repositories:
  - name: backups
    remote_url: "https://github.com/nautobot/demo-gc-backups"
    branch: main

```

```
Traceback (most recent call last):
  File "/usr/local/bin/nautobot-server", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/nautobot/core/cli/__init__.py", line 54, in main
    run_app(
  File "/usr/local/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/source/design_builder/management/commands/test_render_networktocode_designs.py", line 69, in handle
    job.run(data=None, commit=True)
  File "/source/design_builder/design_builder.py", line 460, in run
    self.implement_design(context, design_file)
  File "/source/design_builder/design_builder.py", line 437, in implement_design
    raise ex
  File "/source/design_builder/design_builder.py", line 434, in implement_design
    self.creator.implement_design(design)
  File "/source/design_builder/object_creator.py", line 430, in implement_design
    self._create_objects(self.model_map[key], value)
  File "/source/design_builder/object_creator.py", line 478, in _create_objects
    CreatorObject(self, model_cls, creator_object)
  File "/source/design_builder/object_creator.py", line 169, in __init__
    self._update_fields()
  File "/source/design_builder/object_creator.py", line 278, in _update_fields
    self.creator.save_model(self)
  File "/source/design_builder/object_creator.py", line 506, in save_model
    model.instance.save()
  File "/usr/local/lib/python3.8/site-packages/nautobot/extras/models/datasources.py", line 125, in save
    transaction.on_commit(on_commit_callback)
  File "/usr/local/lib/python3.8/site-packages/django/db/transaction.py", line 128, in on_commit
    get_connection(using).on_commit(func)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 645, in on_commit
    func()
  File "/usr/local/lib/python3.8/site-packages/nautobot/extras/models/datasources.py", line 109, in on_commit_callback
    assert self.request is not None, "No HTTP request associated with this update!"
AssertionError: No HTTP request associated with this update!
```

--- a/docs/admin/release_notes/version_1.1.md
+++ b/docs/admin/release_notes/version_1.1.md
@@ -4,14 +4,16 @@
 
 - 
 
-
 ## [v1.6.0] - 2023-09
 
 ### Added
 
-- Added
+- Added `deferred` attribute to allow deferral of field assignment. See notes in the Changed section.
+
+- Added `model_metadata` attribute to models. At the moment, this provides the ability to specify additional arguments passed to the `save` method of models being updated. The known use case for this is in creating Git repositories in Nautobot 1.x where `trigger_resync` must be `False`. In the future, additional fields will be added to `model_metadata` to provide new functionality.
 
 ### Changed
 
-- Added `deferred` attribute to allow deferral of field assignment.
 - Renamed `nautobot_design_builder.design.Builder` to `nautobot_design_builder.Environment` - aliased original name with deprecation warning.
+
+- Any designs that set `OneToOne` relationships (such as device `primary_ip4`) may now need a `deferred: true` statement in their design for those fields. Previously, `OneToOne` relationships were always deferred and this is usually unnecessary. Any deferrals must now be explicit.

--- a/nautobot_design_builder/design.py
+++ b/nautobot_design_builder/design.py
@@ -122,7 +122,7 @@ class ModelMetadata:  # pylint: disable=too-many-instance-attributes
 
     ACTION_CHOICES = [GET, CREATE, UPDATE, CREATE_OR_UPDATE]
 
-    def __init__(self, model_instance: "ModelInstance"):
+    def __init__(self, model_instance: "ModelInstance", **kwargs):
         """Initialize the metadata object for a given model instance.
 
         By default, the metadata object doesn't really have anything in it. In order
@@ -142,6 +142,8 @@ class ModelMetadata:  # pylint: disable=too-many-instance-attributes
             self.POST_INSTANCE_SAVE: [],
             self.POST_SAVE: [],
         }
+
+        self.save_args = kwargs.get("save_args", {})
 
         # The following attributes are dunder attributes
         # because they should only be set in the @attributes.setter
@@ -425,7 +427,7 @@ class ModelInstance:
         """
         self.environment = environment
         self.instance: Model = None
-        self.metadata = ModelMetadata(self)
+        self.metadata = ModelMetadata(self, **attributes.pop("model_metadata", {}))
 
         self._parent = parent
         self.refresh_custom_relationships()
@@ -597,7 +599,7 @@ class ModelInstance:
         msg = "Created" if self.metadata.created else "Updated"
         try:
             self.instance.full_clean()
-            self.instance.save()
+            self.instance.save(**self.metadata.save_args)
             self.metadata.created = False
             if self._parent is None:
                 self.environment.log_success(

--- a/nautobot_design_builder/tests/test_builder.py
+++ b/nautobot_design_builder/tests/test_builder.py
@@ -124,12 +124,13 @@ def builder_test_case(data_dir):
                     for extension in testcase.get("extensions", []):
                         extensions.append(_load_class(extension))
 
-                    for design in testcase["designs"]:
-                        environment = Environment(extensions=extensions)
-                        commit = design.pop("commit", True)
-                        environment.implement_design(design=design, commit=commit)
-                        if not commit:
-                            roll_back.assert_called()
+                    with self.captureOnCommitCallbacks(execute=True):
+                        for design in testcase["designs"]:
+                            environment = Environment(extensions=extensions)
+                            commit = design.pop("commit", True)
+                            environment.implement_design(design=design, commit=commit)
+                            if not commit:
+                                roll_back.assert_called()
 
                     for index, check in enumerate(testcase.get("checks", [])):
                         for check_name, args in check.items():

--- a/nautobot_design_builder/tests/testdata/nautobot_v1/git_repo.yaml
+++ b/nautobot_design_builder/tests/testdata/nautobot_v1/git_repo.yaml
@@ -1,0 +1,14 @@
+---
+designs:
+  - git_repositories:
+      - model_metadata:
+          save_args:
+            trigger_resync: false
+        name: "backups"
+        remote_url: "https://github.com/nautobot/demo-gc-backups"
+        branch: "main"
+
+checks:
+  - model_exists:
+      model: "nautobot.extras.models.GitRepository"
+      query: {name: "backups"}

--- a/nautobot_design_builder/tests/testdata/nautobot_v2/git_repo.yaml
+++ b/nautobot_design_builder/tests/testdata/nautobot_v2/git_repo.yaml
@@ -1,0 +1,11 @@
+---
+designs:
+  - git_repositories:
+      - name: "backups"
+        remote_url: "https://github.com/nautobot/demo-gc-backups"
+        branch: "main"
+
+checks:
+  - model_exists:
+      model: "nautobot.extras.models.GitRepository"
+      query: {name: "backups"}

